Wrappers/Python/test/test_out_in_place.py

```diff
@@ -313,3 +315,132 @@ def test_proximal_out(self):
                 result=self.get_result(operator, 'adjoint', data)
                 self.out_test(result, operator, 'adjoint',  data)
                 self.in_place_test(result,operator,  'adjoint',  data)
+
+class TestProcessorOutandInPlace(CCPiTestClass):
+    def setUp(self):
+        
+        self.data_arrays=[np.random.normal(0,1, (10,20)).astype(np.float32),  
+                          np.array(range(0,65500, 328), dtype=np.uint16).reshape((10,20)),
+                          np.random.uniform(0,1,(10,20)).astype(np.float32)]
+
+        ag_parallel_2D = AcquisitionGeometry.create_Parallel2D()
+        angles = np.linspace(0, 360, 10, dtype=np.float32)
+        ag_parallel_2D.set_angles(angles)
+        ag_parallel_2D.set_panel(20)
+
+        ag_parallel_3D = AcquisitionGeometry.create_Parallel3D()
+        ag_parallel_3D.set_angles(angles)
+        ag_parallel_3D.set_panel([20,2])
+
+        ag_cone_2D = AcquisitionGeometry.create_Cone2D(source_position=[0,-10],detector_position=[0,10])
+        ag_cone_2D.set_angles(angles)
+        ag_cone_2D.set_panel(20)
+
+        ag_cone_3D = AcquisitionGeometry.create_Cone3D(source_position=[0,-10,0],detector_position=[0,10,0])
+        ag_cone_3D.set_angles(angles)
+        ag_cone_3D.set_panel([20,2])
+        
+        self.geometry_test_list = [ag_parallel_2D, ag_parallel_3D, ag_cone_2D, ag_cone_3D]
+        self.data_test_list= [geom.allocate(None) for geom in self.geometry_test_list]
+
+
+    def out_tests(self, processor, data, *args):
+        input=data.copy() #To check that it isn't changed after function calls
+
+        try:
+            processor.set_input(data)
+            out1 = processor.get_output()
+            self.assertDataArraysInContainerAllClose(input, data, rtol=1e-5, msg= "In case processor.set_input(data), processor.get_output() where processor is  " + processor.__class__.__name__+ " the input data has been incorrectly affected by the calculation.")
+        except NotImplementedError:
+            print("get_result test not implemented for  " + processor.__class__.__name__)
```

Comment:
I think this should force a failure here, maybe `self.fail( err_msg)`

Reply:
Good idea, added this now